flink-runtime/src/test/java/org/apache/flink/runtime/state/TaskStateManagerImplTest.java

```diff
@@ -148,22 +150,22 @@ public void testForwardingSubtaskLocalStateBaseDirFromLocalStateStore() throws I
 				taskLocalStateStore);
 
 			LocalRecoveryDirectoryProvider directoryProviderFromTaskLocalStateStore =
-				taskLocalStateStore.createLocalRecoveryRootDirectoryProvider();
+				taskLocalStateStore.getLocalRecoveryRootDirectoryProvider();
 
 			LocalRecoveryDirectoryProvider directoryProviderFromTaskStateManager =
 				taskStateManager.createLocalRecoveryRootDirectoryProvider();
 
 
 			for (int i = 0; i < 10; ++i) {
 				Assert.assertEquals(rootDirs[i % rootDirs.length],
-					directoryProviderFromTaskLocalStateStore.nextRootDirectory());
+					directoryProviderFromTaskLocalStateStore.rootDirectory(i));
 				Assert.assertEquals(rootDirs[i % rootDirs.length],
-					directoryProviderFromTaskStateManager.nextRootDirectory());
+					directoryProviderFromTaskStateManager.rootDirectory(i));
 			}
 
-			Assert.assertEquals(
-				directoryProviderFromTaskLocalStateStore.getSubtaskSpecificPath(),
-				directoryProviderFromTaskStateManager.getSubtaskSpecificPath());
+//			Assert.assertEquals(
+//				directoryProviderFromTaskLocalStateStore.getSubtaskSpecificPath(),
+//				directoryProviderFromTaskStateManager.getSubtaskSpecificPath());
```

Comment:
Why did you comment this part out? Can it be removed?

Reply:
removed in later commit